### PR TITLE
Add dark mode styles to API pagination table

### PR DESCRIPTION
**Pull request by KATIETOLER**
## Description & motivation 💭 
DT-1982
Add dark mode styles to API pagination table
 

Screenshots:

Screenshot 2024-04-03 at 11 53 53 AM

### CODE REVIEW

File: src/lib/holocene/table/paginated-table.svelte [hunk 1]
```diff
@@ -180,7 +180,9 @@
     @apply surface-primary;
 
     :global(tr:not(.empty)) {
-      @apply h-12 border-b border-primary last-of-type:border-0 hover:bg-gradient-to-br hover:from-blue-100 hover:to-purple-100 hover:bg-fixed;
+      @apply h-12 border-b border-table last-of-type:border-0;
+      @apply dark:hover:bg-gradient-to-br dark:hover:from-indigo-100 dark:hover:to-indigo-700 dark:hover:bg-fixed dark:hover:text-inverse;
+      @apply hover:surface-subtle dark:hover:text-inverse;
```
Comment (laurakwhit) [inline src/lib/holocene/table/paginated-table.svelte:185]: Can these be combined? 

```suggestion
      @apply h-12 border-b border-table hover:surface-subtle last-of-type:border-0 dark:hover:bg-gradient-to-br dark:hover:from-indigo-100 dark:hover:to-indigo-700 dark:hover:bg-fixed dark:hover:text-inverse dark:hover:text-inverse;
```

**Review by laurakwhit — COMMENTED** [review]
Looks like there are a couple of other components that are used in the Workflows paginated table that maybe we want to update in this PR.

First one is the `BulkActionButton` here
Screenshot 2024-04-05 at 11 05 59 AM

And the second one is the `FilterOrCopyButtons` which could use a hover state in dark mode
Screenshot 2024-04-05 at 11 07 00 AM

 If you'd prefer to do a follow-up PR that's fine too! 

**Note:** Looks like you included a screenshot of the `Schedules` table in the PR description, but wanted to note that that is currently not using the `PaginatedTable` component (only the `Workflows` table is).

File: src/lib/holocene/table/paginated-table.svelte (hunk 1 above)
Comment (KATIETOLER) [reply src/lib/holocene/table/paginated-table.svelte:185]: Sure! I changed this to be more human-readable but I will change it back!

File: src/lib/holocene/menu/menu-button.svelte [hunk 2]
```diff
@@ -174,7 +174,7 @@
   }
 
   .table-header {
-    @apply h-auto max-w-fit border-2 border-transparent bg-primary p-0 text-sm text-white focus-visible:border-white;
+    @apply h-auto max-w-fit border-2 border-transparent bg-primary p-0 text-sm text-inverse focus-visible:border-white dark:bg-interactive;
   }
```
Comment (laurakwhit) [inline src/lib/holocene/menu/menu-button.svelte:178]: Fixed in https://github.com/temporalio/ui/pull/1984.

File: src/lib/holocene/filter-or-copy-buttons.svelte [hunk 3]
```diff
@@ -61,7 +61,7 @@
   }
 
   .copy-or-filter-button {
-    @apply surface-primary relative top-[50%] h-fit translate-y-[-50%] rounded-full p-1 text-primary hover:bg-inverse hover:text-white;
+    @apply relative top-[50%] h-fit translate-y-[-50%] rounded-full p-1 text-primary hover:bg-inverse hover:text-white;
```
Comment (laurakwhit) [inline src/lib/holocene/filter-or-copy-buttons.svelte:64]: Will tackle separately as part of `DT-2007`.

**Review by laurakwhit — COMMENTED** [review]
| Before | After |
|--|--|
|Screenshot 2024-04-10 at 3 31 16 PM|Screenshot 2024-04-10 at 3 31 33 PM|
|Screenshot 2024-04-10 at 3 30 56 PM|Screenshot 2024-04-10 at 3 30 38 PM|
|Screenshot 2024-04-10 at 3 31 07 PM|Screenshot 2024-04-10 at 3 30 23 PM|

**Review by laurakwhit — APPROVED** [review]

**Pull request merged**